framework/auth/views.py

```diff
@@ -855,6 +864,17 @@ def external_login_email_post():
     external_id_provider = session.data['auth_user_external_id_provider']
     external_id = session.data['auth_user_external_id']
     fullname = session.data['auth_user_fullname']
+    service_url = session.data['service_url']
+
+    destination = 'dashboard'
+    for campaign in campaigns.CAMPAIGNS:
+        if campaign != 'institution':
+            campaign_url = furl.furl(campaigns.campaign_url_for(campaign)).url
+            if campaigns.is_proxy_login(campaign):
```

Comment:
For CR: handle proxied campaigns such as ember-preprints and future branded ones.